checker/src/main/java/org/checkerframework/checker/calledmethods/builder/LombokSupport.java

```diff
@@ -0,0 +1,204 @@
+     *
+     * @param t a method that is possibly the {@code build} method for a builder. The only
+     *     guaranteed condition is that the enclosing class for the method is itself an inner class.
+     */
+    @Override
+    public void handlePossibleBuilderBuildMethod(AnnotatedExecutableType t) {
+        ExecutableElement element = t.getElement();
+
+        if (isBuilderBuildMethod(element)) {
+            TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+            Element nextEnclosingElement = enclosingElement.getEnclosingElement();
+
+            List<String> requiredProperties = getLombokRequiredProperties(nextEnclosingElement);
+            AnnotationMirror newCalledMethodsAnno =
+                    atypeFactory.createAccumulatorAnnotation(requiredProperties);
+            t.getReceiverType().addAnnotation(newCalledMethodsAnno);
+        }
+    }
+
+    @Override
+    public void handlePossibleToBuilder(AnnotatedExecutableType t) {
+
+        AnnotatedTypeMirror returnType = t.getReturnType();
+        ExecutableElement element = t.getElement();
+
+        String methodName = element.getSimpleName().toString();
+
+        // make sure the method is toBuilder
+        if ("toBuilder".equals(methodName)) {
+            TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+
+            if (ElementUtils.hasAnnotation(element, "lombok.Generated")
+                    || ElementUtils.hasAnnotation(enclosingElement, "lombok.Generated")) {
+                handleToBuilderType(returnType, enclosingElement);
+            }
+        }
+    }
+
+    /**
+     * Add, to a type, a CalledMethods annotation that states that all required setters have been
+     * called. The type can be the return type of toBuilder or the corresponding generated "copy"
+     * constructor.
+     *
+     * @param type type to update
+     * @param classElement corresponding AutoValue class
+     */
+    private void handleToBuilderType(AnnotatedTypeMirror type, Element classElement) {
+        List<String> requiredProperties = getLombokRequiredProperties(classElement);
+        AnnotationMirror calledMethodsAnno =
+                atypeFactory.createAccumulatorAnnotation(requiredProperties);
+        type.replaceAnnotation(calledMethodsAnno);
+    }
+
+    /**
+     * Computes the required properties of a @lombok.Builder class, i.e., the names of the fields
+     * with @lombok.NonNull annotations.
+     *
+     * @param lombokClassElement the class with the @lombok.Builder annotation
+     * @return a list of required property names
+     */
+    private List<String> getLombokRequiredProperties(final Element lombokClassElement) {
+        List<String> requiredPropertyNames = new ArrayList<>();
+        List<String> defaultedPropertyNames = new ArrayList<>();
+        for (Element member : lombokClassElement.getEnclosedElements()) {
+            if (member.getKind() == ElementKind.FIELD) {
```

Comment:
Lombok never generates fields with initializers in classes that use `@Builder`, unless that field has an `@Default` or `@Singular` annotation, which are handled elsewhere (and don't need to be included here, because they can never be null).

Reply:
I'll add a comment to this effect.